src/Symfony/Component/Notifier/Bridge/SmsBiuras/SmsBiurasTransport.php

```diff
@@ -0,0 +1,111 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\SmsBiuras;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Vasilij Duško <vasilij@prado.lt>
+ */
+final class SmsBiurasTransport extends AbstractTransport
+{
+    protected const HOST = 'savitarna.smsbiuras.lt';
+
+    private $uid;
+    private $apiKey;
+    private $from;
+    private $test;
+
+    private const ERROR_CODES = [
+        1 => 'The message was processed and sent to the mobile operator. But delivery confirmations have not yet been returned.',
+        2 => 'SMS not delivered.',
+        3 => 'The SMS message was successfully delivered to the recipient.',
+        4 => 'The message was sent and expired because it could not be delivered to the recipient during its validity period (48 hours according to our default platform).',
+        5 => 'The message was received but the operator returned "Rejected" as the final status.',
+        6 => 'Missing parameters, check that you are using all required parameters.',
+        7 => ' Wrong apikey or uid.',
+        8 => 'Sender ID - "from". Must be approved by an administrator.',
+        9 => 'Balance insufficient, please top up the account.',
+        10 => 'Bad date format for schedule parameter. Ex: urlencode("2021-03-11 12:00").',
+        999 => 'Unknown Error',
+    ];
+
+    public function __construct(string $uid, string $apiKey, string $from, bool $test, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->uid = $uid;
+        $this->apiKey = $apiKey;
+        $this->from = $from;
+        $this->test = $test;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('smsbiuras://%s?from=%s&test=%s', $this->getEndpoint(), $this->from, $this->test);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = 'https://'.$this->getEndpoint().'/api?'.http_build_query([
```

Comment:
use `$response = $this->client->request('GET', $endpoint, ['query' => [....]])` instead of building the URL